tests/unit/test_bam_utils.py

```diff
@@ -106,3 +157,59 @@ def test_bam_header_missing_model(bam_header_without_program_group):
     assert ret == False
 
 
+@pytest.mark.parametrize("bam_header_with_multiple_program_groups", ['mas15teloprimev2'], indirect=True)
+def test_load_models_from_bam_header(bam_header_with_multiple_program_groups):
+    with tempfile.NamedTemporaryFile(delete=True) as f:
+        with pysam.AlignmentFile(f.name, "wb", header=bam_header_with_multiple_program_groups) as bf:
+            bf.close()
+
+        models = bam_utils.load_models(None, input_bam=f)
+
+        assert len(models) == 2
+        assert models[0].name == 'mas15v2'
+        assert models[1].name == 'mas15teloprimev2'
+
+
+def test_load_model_from_name():
+    for model_name in list(model.LibraryModel.pre_configured_models.keys()):
+        lb_models = bam_utils.load_models([model_name])
+
+        assert lb_models[0].name == model_name
```

Comment:
I think in general we should do a test for equality on the json serialized version of the models (or on the objects themselves).  It's more thorough and will ensure we stay up-to-date when we propagate changes.

Reply:
Fixed.